### DISCUSSION TITLE
pdf should be how to replace the text in pdf, but do not change the original layout, only add or delete two words, in GitHub for a long time did not find, why

### DISCUSSION THREAD

**Discussion by 2622594863** [py-pdf/pypdf#2716] [Q&A]
## Explanation

pdf should be how to replace the text in pdf, but do not change the original layout, only add or delete two words, in GitHub for a long time did not find, why

**Comment by stefan6419846** [comment]
This is not trivial and highly depends on the actual PDF file. There have been some discussions about this in #2489 already, which seems to be roughly the same use-case.

**Reply by pubpub-zz** [reply]
pdf is an "e-paper" format. Sometimes printing can be line per line, some times words per word, or even character per character.

**Reply by hoehermann** [reply]
I have been working on text replacement here: https://github.com/hoehermann/pypdf_strreplace It is hit or miss but can be useful.
